### PR TITLE
Rename OP_NOP2 to OP_CHECKLOCKTIMEVERIFY

### PR DESCRIPTION
**Pull request by mb300sd**
As suggested by petertodd here https://www.reddit.com/r/Bitcoin/comments/3wt1b9/bip65_cltv_just_went_into_full_enforcement/cxywon5

:)

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK

Could  you please squash those commits? Commits should be as small as possible but still [atomic](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#contributor-workflow).

**Comment by petertodd** [comment]
It'd be a good idea to note that the name has changed in the release notes, as this has the (small) potential to break some code that depends on what decodescript returns; we probably won't release this until v0.13

File: qa/rpc-tests/test_framework/script.py [hunk 1]
```diff
@@ -226,7 +226,8 @@ def __new__(cls, n):
 
 # expansion
 OP_NOP1 = CScriptOp(0xb0)
-OP_NOP2 = CScriptOp(0xb1)
+OP_CHECKLOCKTIMEVERIFY = CScriptOp(0xb1)
+OP_NOP2 = OP_CHECKLOCKTIMEVERIFY
```
Comment (luke-jr) [inline qa/rpc-tests/test_framework/script.py]: I am not sure this "compatibility" should be retained. If someone is specifying OP_NOP2, they may actually be expecting OP_NOP2 behaviour. Better to error and let them explicitly change to OP_CHECKLOCKTIMEVERIFY.

**Comment by luke-jr** [comment]
@petertodd If they depend on OP_NOP2 being returned, they are likely to be more subtley broken by *not* updating this...

**Comment by dcousens** [comment]
Agreed with @luke-jr, the meaning has changed, a break would be the best-case result for anyone depending on this.

**Comment by mb300sd** [comment]
Updated with release note update and the compatibility define removed.

**Comment by dcousens** [comment]
concept/once-over utACK 0bf46e0

**Comment by mb300sd** [comment]
@paveljanik Only if the verbose option is specified, the json output contains an asm script field. Standard hex output is not.

**Comment by paveljanik** [comment]
@mb300sd Right. Thank you. Can you please add "(in verbose mode)"?

**Comment by MarcoFalke** [comment]
utACK 37d271d

**Comment by petertodd** [comment]
@luke-jr That's a reasonable idea; let's remove OP_NOP2.

Release notes look good, thanks!

**Comment by laanwj** [comment]
utACK (but the tests are sufficient here)
37d271d7cce44885f835292ffe99b54399b014d6

**Comment by maaku** [comment]
tested ACK

**Pull request merged**